bson/src/main/org/bson/codecs/configuration/mapper/MethodModel.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2008-2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs.configuration.mapper;
+
+import com.fasterxml.classmate.members.ResolvedMethod;
+import org.bson.codecs.configuration.CodecRegistry;
+
+/**
+ * Represents the metamodel of a method on a class mapped by a ClassModel
+ */
+public class MethodModel {
```

Comment:
It's a bit of a smell that i haven't quite mitigated.  It initially did but MappedType has a getType() and the closest a method has to a type is its return type.  I think I changed it when I hit an NPE trying to extract the class of the return type on a void method.  I can *probably* fix it but I just haven't come back around to this one yet.  Ideally, yes, it'd be MappedType as well.